### ISSUE TITLE
No titles are really needed for announcments

### ISSUE THREAD

**Issue by iafan** [translate/pootle#4179]
Currently we force announcement pages to have titles, though they are not strictly necessary: we never display announcements outside of the page context; having a header in the sidebar is not strictly needed, so why force people fill it out? If they need to, they can add any headings in the content itself.

Getting rid of the titles makes it slightly easier for people to work with announcements, and would simplify the editing UI (see #3984).

Instead of the user-defined titles, we can display some subtle heading bars saying something generic, like "Language Announcements", "Project Announcement", "Translation Project Announcement" for the sole purpose of telling apart multiple announcement levels in the context of TP sidebars, where we can have mulitple announcements combined. I'll provide a mockup for this later.

**Comment by iafan** [comment]
We can also probably just write the project and language name in such headings (since the word 'Announcements' might be too narrow, and some people may want to put pretty static content there, not strictly announcements). This will also help people understand which level this content belongs to.

---

Here's the example based on http://mozilla.locamotion.org/az/firefox/

Before (big `Project instructions` page title):

![screenclip 2](https://cloud.githubusercontent.com/assets/1728158/11130336/56392862-8939-11e5-8b31-d27eded89ae5.png)

After (small `Project: Firefox 44 (Alpha)` heading):

![screenclip 1](https://cloud.githubusercontent.com/assets/1728158/11130375/7ada1c9e-8939-11e5-9d3a-c6bf50cc721b.png)

**Comment by unho** [comment]
Announcements are actually a special type of `StaticPage` which extends `AbstractPage`. The `title` is defined in the `AbstractPage` model. There is also a `LegalPage` that extends `AbstractPage`.

So should `StaticPage` and `LegalPage` keep the `title`?

See https://github.com/translate/pootle/blob/master/pootle/apps/staticpages/models.py for reference.

**Comment by julen** [comment]
I was about to mention some historical notes on announcements. We added these on top of static pages because we needed a quick way to display some content (as an announcement) without making further changes to the underlying models or doing extra UI/view work for admins. From the very beginning we totally considered this to be a hack — which has worked fine, however it has its own shortcomings as you already noticed.

So I wouldn't say titles are the only redundant part for announcements, but also the redirect field. At the same time, the virtual path is an implementation detail that currently leaks in the UI. From the site admins' perspective the only thing they want is to assign an announcement to a context: a specific language, project or TP. And even more, this is a detail which has to be hidden when editing announcements in their natural context (#3984).

**Comment by unho** [comment]
So, to my understanding, `Announcement` is better to be refactored out of `StaticPage` model to have one of its own. The only really required data for an announcement are the `body`, `active` status and the `virtual_path` (to keep the reference to the TP, Project or Language they are attached to). So `redirect URL` and `title` are fields that are actually not necessary for announcements. The upside of this refactoring is simplifying the forms for editing announcements and remove some ugly hacks on `StaticPage` model. I am not sure about the implications on the views code, but in the admin site announcements are displayed on a separate section. Am I right about all this?

@phlax The optional title can be achieved by including it on the body, as pointed out in the very first comment for this issue.

**Comment by phlax** [comment]
@unho - i understand that - but it doesnt *encourage* users to use have a properly formatted header

**Comment by unho** [comment]
Announcements are displayed in context, and we are adding in context editing, so as @iafan pointed out having a title will no longer be actually necessary.

**Comment by iafan** [comment]
> i understand that - but it doesnt encourage users to use have a properly formatted header

The point is that there shouldn't be a 'properly formatted header' (title, that is), since announcements should look nice without a title. There will be a header that shows the scope of the announcement, though (see the mockup screenshots in the case).

**Comment by phlax** [comment]
hi @iafan - im not suggesting there should be an h1 header - which is what im understanding you mean by a `title` - but as i said above from a semantic pov its better to have some kind of header - and for that kind of header to be standard in these sidebar sections.

**Comment by iafan** [comment]
> So should StaticPage and LegalPage keep the title?

@unho, absolutely. They are displayed in the context of their own pages (URLs), and separate title is needed to make sure we display something meaningful in HTML `<title>` tag.

**Comment by iafan** [comment]
> but as i said above from a semantic pov its better to have some kind of header - and for that kind of header to be standard in these sidebar sections.

@phlax then this is already a part of the proposal.
